src/Sylius/Bundle/CoreBundle/Checkout/Step/FinalizeStep.php

```diff
@@ -89,5 +89,7 @@ protected function completeOrder(OrderInterface $order)
 
         $this->dispatchCheckoutEvent(SyliusCheckoutEvents::FINALIZE_COMPLETE, $order);
         $this->dispatchCheckoutEvent(SyliusOrderEvents::POST_CREATE, $order);
+
+        $this->getCartProvider()->abandonCart();
```

Comment:
IMHO we are not abandoning a cart. It can be misleading. Even if logic is correct, we should rename this step. `lockCart`/`closeCart`?